File: bddtests/src/main/kotlin/mocking/tpp/models/Application.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute

@XmlAccessorType(XmlAccessType.FIELD)
data class Application(
        @XmlAttribute var name: String = "default name",
        @XmlAttribute var version: String = "default version",
        @XmlAttribute var providerId: String = "default providerId",
        @XmlAttribute var deviceType: String = "default deviceType"
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/Application.cs
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    public sealed class Application
    {
        [XmlAttribute("name")]
        public string? Name { get; set; }

        [XmlAttribute("version")]
        public string? Version { get; set; }

        [XmlAttribute("providerId")]
        public string? ProviderId { get; set; }

        [XmlAttribute("deviceType")]
        public string? DeviceType { get; set; }
    }
}